drivers/ipm/ipm_mcux.c

```diff
@@ -0,0 +1,176 @@
+/*
+ * Copyright (c) 2017-2018, NXP
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <errno.h>
+#include <device.h>
+#include <ipm.h>
+#include <fsl_mailbox.h>
+#include <fsl_clock.h>
+#include <soc.h>
+
+#define MCUX_IPM_DATA_REGS 1
+#define MCUX_IPM_MAX_ID_VAL 0
+
+#if defined(__CM4_CMSIS_VERSION)
+#define MAILBOX_ID_THIS_CPU kMAILBOX_CM4
+#define MAILBOX_ID_OTHER_CPU kMAILBOX_CM0Plus
+#else
+#define MAILBOX_ID_THIS_CPU kMAILBOX_CM0Plus
+#define MAILBOX_ID_OTHER_CPU kMAILBOX_CM4
+#endif
+
+
+struct mcux_mailbox_config {
+	MAILBOX_Type *base;
+	void (*irq_config_func)(struct device *dev);
+};
+
+struct mcux_mailbox_data {
+	ipm_callback_t callback;
+	void *callback_ctx;
+};
+
+static void mcux_mailbox_isr(void *arg)
+{
+	struct device *dev = arg;
+	struct mcux_mailbox_data *data = dev->driver_data;
+	mailbox_cpu_id_t cpu_id;
+
+	cpu_id = MAILBOX_ID_THIS_CPU;
+
+	volatile u32_t value = MAILBOX_GetValue(MAILBOX, cpu_id);
```

Comment:
Get the base address from the device config. There are a few other places in the driver that need the same change

Reply:
Done, fixed to use base from config struct